accord-core/src/main/java/accord/coordinate/Invalidate.java

```diff
@@ -113,33 +119,35 @@ public synchronized void onSuccess(Id from, InvalidateReply reply)
 
         InvalidateOk ok = (InvalidateOk) reply;
         invalidateOks.add(ok);
-        if (preacceptTracker.success(from))
+        if (tracker.onSuccess(from) == Success)
             invalidate();
     }
 
     private void invalidate()
     {
+        isBallotPromised = true;
```

Comment:
This is because we may have to _propose_ an invalidation, and we want to separately say we've finished the invalidation, and we've finished the preparation for taking our next step.

But you're right, isDone (and in fact isBallotPromised) should both be redundant. We can remove them if you like. Or at least add an Invariants check